api/hiveextension/v1beta1/agentclusterinstall_types.go

```diff
@@ -329,6 +341,21 @@ const (
 	WorkerAgentMachinePool string = "worker"
 )
 
+// PlatformType is a specific supported infrastructure provider.
+// +kubebuilder:validation:Enum="";BareMetal;None;VSphere
```

Comment:
@avishayt if i remember correctly we wanted to avoid enums no?

Reply:
i don't remember the reason, just remember that we talked about it and somehow changes here are considered breaking changes or something like that

Reply:
The problem with enums is when reporting status because adding a new value breaks clients (e.g., a client checking progress in a state machine and all of the sudden a new state is added).  However, in the spec it should be OK because if we add a new value, it wouldn't affect the others.

Reply:
but the same type is used in the spec and in the status here

Reply:
That's a good point.  The cleanest API would create several conditions: PlatformBareMetal, PlatformNone, PlatformVsphere, and UserManagedNetworking.  Maybe overkill though?

Reply:
Won't it break clients as well if we drop support for a platform? Or we'll have to keep maintaining separate conditions also for deprecated platforms :/ Any way, not sure a scenario of a state machine is relevant here, as the platform in status is simply derived from the spec.

Reply:
I think it's fine the way it is